proposals/apple-mdm/base-setup-clix.md

```diff
@@ -0,0 +1,205 @@
+Generating MDM keys at ./
+  fleet-apns-pki.crt ... done
+  fleet-apns-pki.key ... done
+  fleet-apns-push.key ... done
+  fleet-apns-push.csr ... done
+Send the .csr and .crt files to your friendly Fleet representative
+
+Keep the .key files. You will need them to complete this process. 
+
+Fleet will return a signed certificate to you within 1 business day. Use that file with the .key files in the `fleetctl setup apns gen-req` command.
+```
+
+### Step 2 - Generate .req file from .p7
+``` bash
+$ fleetctl setup apns gen-req \
+--p7-file=~/path/to/*.p7 \
+--pki-key=~/path/to/fleet-apns-pki.key \
+--pki-crt=~/path/to/fleet-apns-pki.crt
+Working...
+Generated fleet-apns-push.req file at ./
+
+Visit identity.apple.com to upload this .req file. For more details, follow these instructions: NEED LINK HERE
+
+Apple will generate a .crt file. Use that file with the .key files in the `fleetctl setup apns finalize` command. 
+```
+
+### Step 3 - Upload push..crt and push.key files to Fleet ENV
+
+```sh
+$ fleetctl setup apns finalize \
+--apple-crt=/path/to/apple.crt \
+--push-key=/path/to/fleetctl-apns-push.key \
+--context=dogfood
+This command will update your Fleet Server's APNS environment variables with the content of the specified files. 
+
+Continue? (Y/n)
+
+Updating ENV["NEED_VARNAME"] with contents of /path/to/apple.crt
+Updating ENV["NEED_VARNAME"] with contents of /path/to/fleetctl-apns-push.key
+```
+
+## SCEP
+### Step 1 - Generate SCEP CA keys
+```bash
+$ fleetctl setup scep-ca init
+Generating SCEP CA and keys at ./
+  fleet-scep.crt ... done
+  fleet-scep.key ... done
+  
+Next, use these file in the `fleetctl setup scep-ca finalize` command.
+```
+
+### Step 2 - Update fleet server ENV vars with new certs
+```bash
+$ fleetctl setup scep-ca finalize \ 
+--scep-crt=/path/to/fleet-scep.crt \
+--scep-key=/path/to/fleet-scep.key \
+--context=dogfood
+This command will update your Fleet Server's SCEP environment variables with the content of the specified files. 
+
+WARNING: If you change the SCEP CA in the fleet server, you must un-enroll and re-enroll all devices.
+
+Continue? (Y/n)
+
+Updating ENV["NEED_VARNAME"] with contents of /path/to/fleet-scep.crt
+Updating ENV["NEED_VARNAME"] with contents of /path/to/fleet-scep.key
+```
+
+## Misc
+### Error States
+#### Missing required flag
+Matches current behavior of `fleetctl`: Simply shows the help page (try `fleetctl setup --name="hello"`)
+#### Required file does not exist 
+Matches current behavior of `fleetctl`: `Error: open does/not/exist: no such file or directory`
+(Try `fleetctl apply -f does/not/exist`)
+### Help pages - Setup
+
+```bash
+$ fleetctl setup
+NAME:
+   fleetctl setup - Set up a Fleet instance or MDM // UPDATED
+
+USAGE:
+   fleetctl setup [options]
+   fleetctl setup apns // NEW
+   fleetctl setup scep // NEW
+   
+SUBCOMMANDS:
+	apns Apple Push Notification Service to enable MDM for Apple devices
+	scep Simple Certificate Enrollment Protocol to enable MDM
```

Comment:
Is there a reason why `scep` help text doesn't include the "...for Apple devices"?

Reply:
Great catch. In theory, SCEP is platform agnostic. We could use it for other OSs... (@GuillaumeRoss @lucasmrod true?)

Reply:
Unknown if Windows MDM protocol uses SCEP... (So far I cannot find good references of SCEP in Microsoft MDM docs: https://docs.microsoft.com/en-us/openspecs/windows_protocols/ms-mdm/33769a92-ac31-47ef-ae7b-dc8501f7104f)

Reply:
Thanks. I'll just say it's for apple MDM then. We can remove that later, if we use it for other platforms.